### ISSUE TITLE
Google Cloud Platform: Workload Identity service account binding

### ISSUE THREAD

**Issue by tisc0** [hashicorp/terraform-provider-google#4765]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Hi,
Actually, Terraform propose already (thanks #3790 I guess) :

- to switch on "workload identity" for cluster
```
workload_identity_config {
    identity_namespace = "${var.google_project}.svc.id.goog"
```

- to switch on "workload identity" for node pool
https://www.terraform.io/docs/providers/google/r/container_cluster.html#workload_identity_config
```
    workload_metadata_config {
      node_metadata = "GKE_METADATA_SERVER"
```
What's missing is bilateral binding between Kube's Service Accounts and GCP IAM Service Accounts.
```
gcloud iam service-accounts add-iam-policy-binding \
  --role roles/iam.workloadIdentityUser \
  --member "serviceAccount:[PROJECT_ID].svc.id.goog[[K8S_NAMESPACE]/[KSA_NAME]]" \
  [GSA_NAME]@[PROJECT_ID].iam.gserviceaccount.com
```
https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity

This would give a complete and industrialized use of Workload Identity, which provides a **secure** and elegant implementation for Service Accounts to:
- access Google's APIs from Kubernetes
- delegating the full management (API keys rotation + access rights) to Google
- avoid the need of extract the JSON keys (cancelling by this way, the automated rotation)
- eliminating one more insecure Kube's secret

In other words, Terraform's module could permit [in a single block ?] to: 
- create Kubes Service Account (KSA) [if non existent]
- create GCP IAM Service Account (GSA) [if non existent]
- define the metadata necessary to "cross-bind" the Service Accounts (Project ID, K8s Namespace)

### New or Affected Resource(s)

Could be:
- google_iam_policy
- kubernetes_service_account_binding ?
- workload_sa_binding ?

### Potential Terraform Configuration

```tf
resource "kubernetes_service_account_binding" "MicroService_1" {
  metadata {
    ksa_name = "$KSA_MS_1"
    gsa_name = "$GSA_MS_1"
    project_id = "$project_id"
    k8s_namespace = "$prod_namespace"
  }

```

### References

* #3790

**Thanks !**

**Comment by morgante** [comment]
You could also take a look at this module which handles things: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/workload-identity

**Comment by rileykarson** [comment]
Based on https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/workload-identity?rgh-link-date=2020-05-12T20%3A21%3A05Z it looks like there aren't any provider-level changes required to support this. Closing.
